### ISSUE TITLE
[idea] As a user of the CLI, I want to know what options/tasks are available

### ISSUE THREAD

**Issue by afeld** [gulpjs/gulp#1826]
Long issue, sorry 😁  Rearranged the sections/formatting a bit to make more sense for the context—hope that's ok. Thanks!

## TL;DR

My suggestions are:

* When someone runs `gulp <non-existent task>`, tell them about the `--tasks` flag to get the list of available tasks.
* Have one or more of the following print out CLI usage documentation.
    * `gulp --help`
    * `gulp -h`
    * `gulp help`

## Backstory

I was trying to build the project I'm helping out on, but didn't know the correct/available Gulp task(s) to do so. I was curious about what the available tasks were, so I looked through the Gulpfile...which was Too Long, so I Didn't Read. I was curious if there was a way to list available tasks, so I tried:

* `gulp tasks`
* `gulp --help`
* `gulp -h`
* `gulp help`

None of these gave me the information I was looking for. I ended up [Googling for "gulpjs list tasks"](https://www.google.com/search?q=gulpjs+list+tasks&oq=gulpjs+list+tasks). The first three results were plugins (which seemed overkill), and only [the fourth result](http://stackoverflow.com/a/30780026) said that there's now a `--tasks` flag available.

### gulp --help

For `gulp --help` and `gulp -h`, it seems to ignore the flag and run the default task.

### gulp help

```
$ gulp help
...
[01:05:37] Task 'help' is not in your gulpfile
[01:05:37] Please check the documentation for proper gulpfile formatting
```

While it's _technically_ correct that the Gulpfile doesn't have a `help` task, someone new to a project (or Gulp, for that matter) might be lost about what to do next.

## What were you expecting to happen?

I would expect one or more of the following:

* `gulp --help`
* `gulp -h`
* `gulp help`

to produce help documentation, such as

```
Usage:

gulp <task>      Runs the specified task.
gulp --tasks     Prints out the list of available tasks.
...

See https://github.com/gulpjs/gulp/blob/master/docs/CLI.md for more information.
```

I hadn't even realized the [CLI documentation page](https://github.com/gulpjs/gulp/blob/master/docs/CLI.md) existed until I searched existing issues...there doesn't seem to be a (direct or transitive) link to it from the [Getting Started](https://github.com/gulpjs/gulp/blob/master/docs/getting-started.md) page.

### gulp non-existent-task

With `gulp help` (or any `gulp <non-existent task>`), I then ran into the problem where I didn't actually know what tasks were available. Gulpfiles aren't always concise enough to see/understand all of the tasks by glancing through the file, so would be helpful to give a little more guidance here. My suggestion:

```
$ gulp <non-existent task>
...
[01:05:37] Task '<non-existent task>' is not in your gulpfile.
[01:05:37] Run 'gulp --tasks' to see the list of available tasks.
...
```

Happy to submit a pull request for this, if it seems agreeable.

## Gulpfile

https://github.com/18F/cg-diagrams/blob/74400adc706440409c21288da4dde9e160d50477/gulpfile.babel.js

## What version of gulp are you using?

3.9.1

## What versions of npm and node are you using?

Node v6.0.0, NPM 3.8.9

**Comment by phated** [comment]
Good news!  https://github.com/gulpjs/gulp-cli already does everything you want.

**Comment by contra** [comment]
We have a ticket open to bring all the docs into the same place too, right now its hard to find docs for the individual pieces (CLI, vinyl-fs, vinyl, etc.).

Any help with that would be much appreciated!

**Comment by contra** [comment]
@phated just to be clear, this is on the 4.x branch of the CLI? Some of the features (showing tasks when an unknown task is used) I haven't seen on 3.x
